src/main/java/uk/ac/ebi/atlas/search/geneids/GeneIdSearchDao.java

```diff
@@ -42,12 +42,13 @@ public GeneIdSearchDao(SolrCloudCollectionProxyFactory collectionProxyFactory) {
     public Optional<ImmutableSet<String>> searchGeneIds(String propertyValue, String propertyName, String species) {
         SolrQueryBuilder<BioentitiesCollectionProxy> bioentitiesQueryBuilder =
                 new SolrQueryBuilder<BioentitiesCollectionProxy>()
-                        .addFilterFieldByTerm(SPECIES, species)
                         .addQueryFieldByTerm(PROPERTY_VALUE, propertyValue)
                         .addQueryFieldByTerm(PROPERTY_NAME, propertyName)
                         .setFieldList(BioentitiesCollectionProxy.BIOENTITY_IDENTIFIER)
                         .sortBy(BioentitiesCollectionProxy.BIOENTITY_IDENTIFIER, SolrQuery.ORDER.asc);
-
+        if (species != "") {
```

Comment:
It’s safer and more readable to use a method such as [`isNotEmpty(species)`](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#isNotEmpty-java.lang.CharSequence-).